browser/src/UI/components/Tabs.tsx

```diff
@@ -46,9 +46,10 @@ export interface ITabsProps {
     visible: boolean
     tabs: ITabProps[]
 
+    shouldUseVimTabs: boolean
```

Comment:
Ideally, the `Tab` component shouldn't need to worry about whether we are using `tabs` or `buffers` - it just gets a bunch of `Tab` objects with `id`, `name`, etc, and hands that back up later.

It looks like the latest fix _only_ impacts `tabs` - what if we move the fix here?
https://github.com/onivim/oni/blob/766fc30133aea41551295c74d458cdec29da6790/browser/src/UI/components/Tabs.tsx#L306

We create the data model for `<Tab />` with either `getTabsFromVimTabs` or `getTabsFromBufferTabs`, depending on the `tabs.mode`.

A minimal fix would be to change `getTabsFromVimTabs` so that it uses `id: t.id + 1` - that way, we wouldn't need a new prop or to change any logic in `<Tab />` (and the buffer strategy stays the same as it is today)

Reply:
That does makes sense. I wasn't too fond of changing the logic all around the file. I'll try and make the changes you suggested.